File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task119.java

import java.io.*;
import java.util.*;

public class Task119 {
    public static List<List<String>> parseCSV(String csvContent) {
        List<List<String>> records = new ArrayList<>();
        if (csvContent == null || csvContent.isEmpty()) {
            return records;
        }
        
        BufferedReader reader = new BufferedReader(new StringReader(csvContent));
        String line;
        
        try {
            while ((line = reader.readLine()) != null) {
                List<String> record = parseLine(line);
                records.add(record);
            }
        } catch (IOException e) {
            System.err.println("Error reading CSV: " + e.getMessage());
        }
        
        return records;
    }
    
    private static List<String> parseLine(String line) {
        List<String> fields = new ArrayList<>();
        StringBuilder field = new StringBuilder();
        boolean inQuotes = false;
        
        for (int i = 0; i < line.length(); i++) {
            char c = line.charAt(i);
            
            if (c == '"') {
                if (inQuotes && i + 1 < line.length() && line.charAt(i + 1) == '"') {
                    field.append('"');
                    i++;
                } else {
                    inQuotes = !inQuotes;
                }
            } else if (c == ',' && !inQuotes) {
                fields.add(field.toString());
                field = new StringBuilder();
            } else {
                field.append(c);
            }
        }
        fields.add(field.toString());
        
        return fields;
    }
    
    public static void printCSV(List<List<String>> records) {
        for (List<String> record : records) {
            System.out.println(record);
        }
    }
    
    public static void main(String[] args) {
        // Test Case 1: Simple CSV
        String csv1 = "Name,Age,City\\nJohn,30,New York\\nJane,25,Los Angeles";
        System.out.println("Test Case 1 - Simple CSV:");
        List<List<String>> result1 = parseCSV(csv1);
        printCSV(result1);
        System.out.println();
        
        // Test Case 2: CSV with quoted fields
        String csv2 = "Name,Description\\n\\"John Doe\\",\\"Software Engineer\\"\\n\\"Jane Smith\\",\\"Data Scientist\\"";
        System.out.println("Test Case 2 - CSV with quoted fields:");
        List<List<String>> result2 = parseCSV(csv2);
        printCSV(result2);
        System.out.println();
        
        // Test Case 3: CSV with commas in quoted fields
        String csv3 = "Product,Price,Description\\nLaptop,1200,\\"High performance, 16GB RAM\\"\\nPhone,800,\\"5G enabled, dual camera\\"";
        System.out.println("Test Case 3 - CSV with commas in quoted fields:");
        List<List<String>> result3 = parseCSV(csv3);
        printCSV(result3);
        System.out.println();
        
        // Test Case 4: CSV with escaped quotes
        String csv4 = "Title,Quote\\nBook,\\"He said \\"\\"Hello\\"\\"\\"\\nMovie,\\"She said \\"\\"Goodbye\\"\\"\\"";
        System.out.println("Test Case 4 - CSV with escaped quotes:");
        List<List<String>> result4 = parseCSV(csv4);
        printCSV(result4);
        System.out.println();
        
        // Test Case 5: Empty and single line CSV
        String csv5 = "Header1,Header2,Header3";
        System.out.println("Test Case 5 - Single line CSV:");
        List<List<String>> result5 = parseCSV(csv5);
        printCSV(result5);
    }
}

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task119.py

def parse_csv(csv_content):
    """\n    Parses CSV content and returns a list of lists.\n    """
    if not csv_content:
        return []
    
    records = []
    lines = csv_content.split('\\n')
    
    for line in lines:
        if line:
            record = parse_line(line)
            records.append(record)
    
    return records


def parse_line(line):
    """\n    Parses a single CSV line, handling quoted fields and escaped quotes.\n    """
    fields = []
    field = []
    in_quotes = False
    i = 0
    
    while i < len(line):
        c = line[i]
        
        if c == '"':
            if in_quotes and i + 1 < len(line) and line[i + 1] == '"':
                field.append('"')
                i += 1
            else:
                in_quotes = not in_quotes
        elif c == ',' and not in_quotes:
            fields.append(''.join(field))
            field = []
        else:
            field.append(c)
        
        i += 1
    
    fields.append(''.join(field))
    return fields


def print_csv(records):
    """\n    Prints CSV records.\n    """
    for record in records:
        print(record)


if __name__ == "__main__":
    # Test Case 1: Simple CSV
    csv1 = "Name,Age,City\\nJohn,30,New York\\nJane,25,Los Angeles"
    print("Test Case 1 - Simple CSV:")
    result1 = parse_csv(csv1)
    print_csv(result1)
    print()
    
    # Test Case 2: CSV with quoted fields
    csv2 = 'Name,Description\\n"John Doe","Software Engineer"\\n"Jane Smith","Data Scientist"'
    print("Test Case 2 - CSV with quoted fields:")
    result2 = parse_csv(csv2)
    print_csv(result2)
    print()
    
    # Test Case 3: CSV with commas in quoted fields
    csv3 = 'Product,Price,Description\\nLaptop,1200,"High performance, 16GB RAM"\\nPhone,800,"5G enabled, dual camera"'
    print("Test Case 3 - CSV with commas in quoted fields:")
    result3 = parse_csv(csv3)
    print_csv(result3)
    print()
    
    # Test Case 4: CSV with escaped quotes
    csv4 = 'Title,Quote\\nBook,"He said ""Hello"""\\nMovie,"She said ""Goodbye"""'
    print("Test Case 4 - CSV with escaped quotes:")
    result4 = parse_csv(csv4)
    print_csv(result4)
    print()
    
    # Test Case 5: Empty and single line CSV
    csv5 = "Header1,Header2,Header3"
    print("Test Case 5 - Single line CSV:")
    result5 = parse_csv(csv5)
    print_csv(result5)
